README.md

```diff
@@ -0,0 +1,209 @@
+- **CartItemRepository**: Interface for CRUD operations on CartItem entities.
+- **OrderRepository**: Interface for CRUD operations on Order entities.
+- **OrderItemRepository**: Interface for CRUD operations on OrderItem entities and methods for retrieving order items by order ID.
+- **UserRepository**: Interface for CRUD operations on User entities and methods for checking if the user is already registered.
+- **RoleRepository**: Interface for CRUD operations on Role entities, including methods for identifying user roles.
+
+## Setup and Usage
+
+To get started with this project, follow these steps:
+
+1. **Clone the repository**:
+    ```sh
+    git clone https://github.com/bodya4243/online-book-store.git
+    cd online-book-store
+    ```
+
+2. **Build the project using Maven**:
+    ```sh
+    mvn clean install
+    ```
+
+3. **Run the application**:
+    ```sh
+    mvn spring-boot:run
+    ```
+   Technologies and Tools
+   Key Technologies
+   Java Development Kit (JDK) 17
+   Maven
+   MySQL
+   Spring Boot
+   H2 Database (for testing)
+   Lombok
+   MapStruct
+   Liquibase
+   SpringDoc OpenAPI
+   JWT (JSON Web Token)
+   Docker Compose
+   
+
+
+4. **Access the application**:
+   - The application runs on: http://localhost:8080.
+   - API documentation is available at: http://localhost:8080/swagger-ui.html.
+
+### Docker Setup
+
+**Swagger**:
+   - You can use Swagger for manipulating API endpoints.
+
+**Docker**:
+   - You can download the container from Docker and start the API.
+
+# MySQL environment variables
+```
+   DB_PASSWORD=123sqlbod321
+   DB_DATABASE=book_store_db
+   DB_USER=bodya
+   DB_LOCAL_PORT=3309
+   DB_DOCKER_PORT=3306
+
+   SPRING_LOCAL_PORT=8080
+   SPRING_DOCKER_PORT=8080
+   DEBUG_PORT=5005
+
+   JWT_SECRET=wefwqfdfwqfq4f3fq3af43453452ef34fewfewfe
+   
+   MYSQL_DATABASE=book_store_db
+   MYSQL_USER=bodya
+   MYSQL_PASSWORD=123sqlbod321
+```
+
+1. **Build the Docker image**:
+    ```sh
+    docker build -t online-book-store .
+    ```
+
+2. **Run the Docker container**:
+    ```sh
+    docker run -d -p 8080:8080 online-book-store
+    ```
+
+3. **Verify the container is running**:
+    - Ensure the application is accessible at `http://localhost:8080`.
+
+4. **Authentication and Usage Guide**
+
+* How It Works
+The application uses JWT (JSON Web Token) for authentication and authorization.
+When a user logs in, they receive a JWT token,
+which must be included in the Authorization header of subsequent requests to access protected endpoints.
+
+* User Registration
+To register a new user, send a POST request to /api/auth/registration with the following JSON payload:
+
+```
+   json
+   {
+      "email": "tom.doe@example.com",
+      "password": "securePassword123",
+      "repeatPassword": "securePassword123",
+      "firstName": "alice",
+      "lastName": "Doe",
+      "shippingAddress": "Kyiv, Shevchenko ave, 1"
+   }
+```
```

Comment:
Same in other json code samples
```suggestion
```json
   {
      "email": "tom.doe@example.com",
      "password": "securePassword123",
      "repeatPassword": "securePassword123",
      "firstName": "alice",
      "lastName": "Doe",
      "shippingAddress": "Kyiv, Shevchenko ave, 1"
   }
```
```